File: 023/main.js
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */

function ListNode (val) {
  this.val = val
  this.next = null
}
var mergeKLists = function (lists) {
  let num = lists.length
  let arr = []
  for (let i = 0; i < num; i++) {
    let item = lists[i]
    while (item != null) {
      arr.push(item.val)
      item = item.next
    }
  }
  arr.sort((a, b) => {
    return a - b > 0 ? 1 : -1
  })
  let res = new ListNode(-1)
  let doc = res
  for (let i = 0; i < arr.length; i++) {
    doc.next = new ListNode(-1)
    doc = doc.next
    doc.val = arr[i]
  }
  return res.next
}

let l1 = {
  val: 2,
  next: {
    val: 4,
    next: {
      val: 3,
      next: null
    }
  }
}

let l2 = {
  val: 5,
  next: {
    val: 6,
    next: {
      val: 4,
      next: null
    }
  }
}

console.log(mergeKLists([l1, l2]))

File: 023/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        vector<int> arr;
        for (int i = 0; i < lists.size(); i++) {
            while (lists[i]) {
                arr.push_back(lists[i]->val);
                lists[i] = lists[i]->next;
            }
        }
        sort(arr.begin(), arr.end());
        ListNode* res = new ListNode(-1);
        ListNode* doc = res;
        for (int i = 0; i < arr.size(); i++) {
            doc->next = new ListNode(-1);
            doc = doc->next;
            doc->val = arr[i];
        }
        return res->next;
    }
};

int main()
{
    Solution sol;
    vector<ListNode*> lists;
    ListNode* l1 = new ListNode(1);
    l1->next = new ListNode(3);
    ListNode* l2 = new ListNode(2);
    l2->next = new ListNode(4);
    lists.push_back(l1);
    lists.push_back(l2);
    ListNode* res = sol.mergeKLists(lists);
    while (res) {
        cout << res->val << endl;
        res = res->next;
    }
    return 0;
}
